components/collator/benches/bench.rs

```diff
@@ -0,0 +1,89 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use std::time::Duration;
+
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+
+use icu::collator::*;
+use icu::locid::Locale;
+
+pub fn collator_with_locale(criterion: &mut Criterion) {
+    // Load file content in reverse order vector.
+    let content_latin: (&str, Vec<&str>) = (
+        "TestNames_Latin",
+        include_str!("data/TestNames_Latin.txt")
+            .split('\n')
+            .rev()
+            .collect::<Vec<&str>>(),
+    );
+    let content_asian: (&str, Vec<&str>) = (
+        "TestNames_Asian",
+        include_str!("data/TestNames_Asian.txt")
+            .split('\n')
+            .rev()
+            .collect(),
+    );
+    let content_russian: (&str, Vec<&str>) = (
+        "TestNames_Russian",
+        include_str!("data/TestNames_Russian.txt")
+            .split('\n')
+            .rev()
+            .collect(),
+    );
+
+    let performance_parameters = [
+        ("en_US", vec![&content_latin]),
+        ("fr_FR", vec![&content_latin]),
+        ("ja_JP", vec![&content_latin, &content_asian]),
+        ("zh_CN", vec![&content_latin]),
+        ("ru_RU", vec![&content_latin, &content_russian]),
+    ];
+
+    for perf_group in performance_parameters {
+        let (locale_under_test, files_under_test) = perf_group;
+        let current_locale: Locale = locale_under_test.parse().expect("Failed to parse locale.");
+
+        let mut options = CollatorOptions::new();
+        options.strength = Some(Strength::Primary);
```

Comment:
Please bench all five strengths. The default is tertiary, so it makes sense to bench that. Benching only primary leaves a bunch on code unbenched. Also, it's particularly interesting to bench quaternary and identical with Japanese due to the performance remarks ICU4C docs make about Japanese. In particular, to get full Japanese standard behavior, you need the identical strength. Furthermore, CLDR used to default to quaternary for Japanese but now defaults to tertiary as for every other language for performance reasons.

Reply:
In addition to making the changes Henri is suggestion, can you also convert Henri's comments into inline source comments? They are very informative for our future selves.

Reply:
Added in the source file and credited Henry.